components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,264 @@
+
+The $z$ location of cell interfaces is found by summing the pseudo thicknesses from the bottom multiplied by the specific volume:
+
+$$ z_{i,k}^{top} = z_i^{floor} + \rho_0 \sum_{k^\prime=k}^{K_{max}} \alpha_{i,k^\prime} \tilde{h}_{i,k^\prime}, $$
+
+where $z_i^{floor}$ is the (positive-up) bottom elevation.
+The $z$ location of cell centers is given by: 
+
+$$ z_{i,k} = z_i^{floor} + \frac{1}{2} \rho_0\alpha_{i,k} \tilde{h}_{i,k} + \rho_0\sum_{k^\prime= k+1}^{K_{max}} \alpha_{i,k^\prime} \tilde{h}_{i,k^\prime}. $$
+
+Initially, the geopotential is the sum of the $z$ height times $g$. 
+In the future, it will include contributions from the tidal potential ($\Phi_{tp}$) and self attraction and loading ($\Phi_{SAL}$):
+
+$$ \Phi_{i,k} = \left( gz_{i,k} + \Phi_{tp} + \Phi_{SAL} \right). $$
+
+The desired pseudo thickness used for the $p^\star$ coordinate is:
+
+$$\tilde{h}_k^{p^\star} = \tilde{h}_k^{rest} + \tilde{h}_k^{p}, $$
+
+where the pressure alteration, $\tilde{h}_k^{p}$, is given by:
+
+$$ \tilde{h}_k^{p} =\frac{(p_B-p_{surf})}{g\rho_0} \frac{W_k\tilde{h}_k^{rest}}{\sum_{k^\prime=1}^K W_{k^\prime}\tilde{h}_{k^\prime}^{rest}}.$$
+
+## 4 Design
+
+### 4.1 Data types and parameters
+
+The `VertCoord` class will be used to compute pressure, $z$ height, and geopotential at each layer.
+It will also contain the vertical loop bounds for cells, edges, and vertices.
+```c++
+class VertCoord {
+    public:
+        I4 NVertLevels;
+        I4 NVertLevelsP1;
+
+        // Variables computed 
+        Array2DReal PressureInterface;
+        Array2DReal PressureMid;        
+        Array2DReal ZInterface;
+        Array2DReal ZMid;
+        Array2DReal GeopotentialMid;
+        Array2DReal LayerThicknessPStar;
+
+        // Vertical loop bounds
+        Array1DI4 MinLevelCell;
+        Array1DI4 MaxLevelCell;
+        Array1DI4 MinLevelEdgeTop;
+        Array1DI4 MaxLevelEdgeTop;
+        Array1DI4 MinLevelEdgeBot;     
+        Array1DI4 MaxLevelEdgeBot;
+        Array1DI4 MinLevelVertexTop;
+        Array1DI4 MaxLevelVertexTop;     
+        Array1DI4 MinLevelVertexBot;
+        Array1DI4 MaxLevelVertexBot;
+
+        // p star coordinate variables
+        Array2DReal VertCoordMovementWeights;
+        Array2DReal RefLayerThickness;
+
+        // Variables from HorzMesh
+        Array2DReal BottomDepth;
+    private:
+
+        // Variables from HorzMesh
+        I4 NCells;
+        I4 NEdges;
+        I4 NVertices;
+        Array2DReal CellsOnEdge;
+        Array2DReal CellsOnVertex;
+
+        static VertCoord *DefaultVertCoord;
+        static std::map<std::string, std::unique_ptr<VertCoord>> AllVertCoords;
+}
+```
+
+### 4.2 Methods
+
+```c++
+class VertCoord{
+    public:
+        static VertCoord *create();
+        void computePressure();
+        void computeZHeight();
+        void computeGeopotential();
+        void computePStarThickness();
```

Comment:
Should we consider generalizing this to `computeTargetPseudoThickness` or something along those lines that's more succinct?

Reply:
That sounds good to me. Maybe just `computeTargetThickness` to cut down on length and since we don't include `Pseudo` in any other variable names (e.g. `layerThickness`)?